firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java

```diff
@@ -1007,90 +1008,148 @@ public void testPartialIndexAndFullIndex() throws Exception {
+    validateIndexType(query15, IndexManager.IndexType.FULL);
   }
 
-  private void validateIsPartialIndex(Query query) {
-    validateIndex(query, false);
-  }
+  @Test
+  public void testIndexTypeForOrQueries() throws Exception {
+    indexManager.addFieldIndex(fieldIndex("coll", "a", Kind.ASCENDING));
+    indexManager.addFieldIndex(fieldIndex("coll", "b", Kind.ASCENDING));
+    indexManager.addFieldIndex(fieldIndex("coll", "b", Kind.ASCENDING, "a", Kind.ASCENDING));
+
+    // OR query without orderBy without limit which has missing sub-target indexes.
+    Query query1 = query("coll").filter(orFilters(filter("a", "==", 1), filter("c", "==", 1)));
+    validateIndexType(query1, IndexManager.IndexType.NONE);
+
+    // OR query with explicit orderBy without limit which has missing sub-target indexes.
+    Query query2 =
+        query("coll")
+            .filter(orFilters(filter("a", "==", 1), filter("c", "==", 1)))
+            .orderBy(orderBy("c"));
+    validateIndexType(query2, IndexManager.IndexType.NONE);
+
+    // OR query with implicit orderBy without limit which has missing sub-target indexes.
+    Query query3 = query("coll").filter(orFilters(filter("a", "==", 1), filter("c", ">", 1)));
+    validateIndexType(query3, IndexManager.IndexType.NONE);
+
+    // OR query with explicit orderBy with limit which has missing sub-target indexes.
+    Query query4 =
+        query("coll")
+            .filter(orFilters(filter("a", "==", 1), filter("c", "==", 1)))
+            .orderBy(orderBy("c"))
+            .limitToFirst(2);
+    validateIndexType(query4, IndexManager.IndexType.NONE);
 
-  private void validateIsFullIndex(Query query) {
-    validateIndex(query, true);
+    // OR query with implicit orderBy with limit which has missing sub-target indexes.
+    Query query5 =
+        query("coll").filter(orFilters(filter("a", "==", 1), filter("c", ">", 1))).limitToFirst(2);
+    validateIndexType(query5, IndexManager.IndexType.NONE);
+
+    // OR query without orderBy without limit which has all sub-target indexes.
+    Query query6 = query("coll").filter(orFilters(filter("a", "==", 1), filter("b", "==", 1)));
+    validateIndexType(query6, IndexManager.IndexType.FULL);
+
+    // OR query with explicit orderBy without limit which has all sub-target indexes.
+    Query query7 =
+        query("coll")
+            .filter(orFilters(filter("a", "==", 1), filter("b", "==", 1)))
+            .orderBy(orderBy("a"));
+    validateIndexType(query7, IndexManager.IndexType.FULL);
+
+    // OR query with implicit orderBy without limit which has all sub-target indexes.
+    Query query8 = query("coll").filter(orFilters(filter("a", ">", 1), filter("b", "==", 1)));
+    validateIndexType(query8, IndexManager.IndexType.FULL);
+
+    // OR query without orderBy with limit which has all sub-target indexes.
+    Query query9 =
+        query("coll").filter(orFilters(filter("a", "==", 1), filter("b", "==", 1))).limitToFirst(2);
```

Comment:
Would recommend mixing limitToLast in some of the cases here. Optional.

Reply:
Done